twigs/ecr.py

```diff
@@ -0,0 +1,172 @@
+    allassets = []
+
+
+    import pdb; pdb.set_trace()
+    if args.repository is None and args.registryId is None:
+        logging.error("Please specify repository name or registryId to be discovered")
+        return None
+    if not args.registryId:
+        
+        logging.info("Starting discovery of repository in ECR")
+
+        if args.repositoryType == 'private':
+            ilist_cmd = "ecr describe-images --repository-name " + args.repository
+        else:
+            ilist_cmd = "ecr-public describe-images --repository-name " + args.repository
+       
+        i_json = run_aws_cmd(ilist_cmd)
+
+        if len(i_json) > 0:
+            args.image = get_image_name_with_tag(args.tag,args.repository,args.repositoryType)
+
+            if args.image is None:
+                logging.info("This repository has no images")
+                return None
+
+            args.assetid = args.image
+
+
+            #docker.py is already taking care of this, do we still need this code?
+            args.assetid = args.assetid.replace('/','-')
+            args.assetid = args.assetid.replace(':','-')
+            args.assetname = args.image
+
+            logging.info("Discovering image "+args.image)
+            assets = docker.get_inventory(args)
+
+            if assets != None:
+                for a in assets:
+                    a['tags'].append('ECR')
+                return assets
+
+        else:
+            logging.error("Repository does not exist, please check the name and run the command again")
+            return None
+    else:
+
+        final_assets= None 
+        import pdb; pdb.set_trace()
+        logging.info("Starting discovery for all repositories under AWS account")
+
+        if args.repositoryType == 'private':
+            ilist_cmd = "ecr describe-repositories"
+        else:
+            ilist_cmd = "ecr-public describe-repositories"
+
+        i_json = run_aws_cmd(ilist_cmd)
+
+
+        if len(i_json) > 0:
+            logging.info("Found %d repositories in %s", len(i_json), args.registryId)
+
+        if i_json['repositories'][0]['registryId'] == args.registryId: 
+            for repos in i_json['repositories']:
+                         
+                args.image = get_image_name_with_tag(args.tag,repos['repositoryName'],args.repositoryType)
+
+                
+                if args.image is not None:
+                    args.assetid = args.image
+                
+                    #docker.py is already taking care of this, do we still need this code?
+                    args.assetid = args.assetid.replace('/','-')
+                    args.assetid = args.assetid.replace(':','-')
+                    args.assetname = args.image
+                
+                    logging.info("Discovering image "+args.image)
+                    assets = docker.get_inventory(args)
+
+                
+                    if assets:
+                        allassets = allassets + assets
```

Comment:
Might be better to use extend(...) i.e. allassets.extend(assets)

Reply:
good choice. done